Realm/RLMAnalytics.mm

```diff
@@ -194,73 +296,206 @@ static auto RLMSysCtl(int *mib, u_int mibSize, size_t *bufferSize) {
+        @"Reference_List": @0,
+        @"Reference_Set": @0,
+        @"Reference_Dictionary": @0,
+    };
+    NSMutableDictionary *featuresDictionary = [[NSMutableDictionary alloc] init];
+    [featuresDictionary addEntriesFromDictionary: dict];
+    for (RLMObjectSchema *objectSchema in schema.objectSchema) {
+        if (objectSchema.isEmbedded) {
+            [featuresDictionary setObject:@1 forKey:@"Embedded_Object"];
+        }
+        if (objectSchema.isAsymmetric) {
+            [featuresDictionary setObject:@1 forKey:@"Asymmetric_Object"];
+        }
+
+        for (RLMProperty *property in objectSchema.properties) {
+            if (property.array) {
+                if (property.type == RLMPropertyTypeObject) {
+                    [featuresDictionary setObject:@1 forKey:@"Reference_List"];
+                } else {
+                    [featuresDictionary setObject:@1 forKey:@"Primitive_List"];
+                }
+                continue;
+            }
+            if (property.set) {
+                if (property.type == RLMPropertyTypeObject) {
+                    [featuresDictionary setObject:@1 forKey:@"Reference_Set"];
+                } else {
+                    [featuresDictionary setObject:@1 forKey:@"Primitive_Set"];
+                }
+                continue;
+            }
+            if (property.dictionary) {
+                if (property.type == RLMPropertyTypeObject) {
+                    [featuresDictionary setObject:@1 forKey:@"Reference_Dictionary"];
+                } else {
+                    [featuresDictionary setObject:@1 forKey:@"Primitive_Dictionary"];
+                }
+                continue;
+            }
+
+            switch (property.type) {
+               case RLMPropertyTypeAny:
+                    [featuresDictionary setObject:@1 forKey:@"Mixed"];
+                  break;
+               case RLMPropertyTypeObject:
+                    [featuresDictionary setObject:@1 forKey:@"Reference_Link"];
+                  break;
+                case RLMPropertyTypeLinkingObjects:
+                     [featuresDictionary setObject:@1 forKey:@"Backlink"];
+                   break;
+               default:
+                    break;
+            }
+        }
+    }
+    return featuresDictionary;
+}
+
+static NSDictionary *RLMConfigurationMetrics(RLMRealmConfiguration *configuration) {
+    BOOL isSyncEnable = configuration.syncConfiguration != nil;
+    BOOL isSync = configuration.syncConfiguration != nil;
+    BOOL isPBSSync = configuration.syncConfiguration.partitionValue != nil;
+    BOOL isFlexibleSync = (isSync && !isPBSSync);
+    BOOL isCompactOnLaunch = configuration.shouldCompactOnLaunch != nil;
+    BOOL migrationBlock = configuration.migrationBlock != nil;
+
+    auto resetMode = configuration.syncConfiguration.clientResetMode;
+    return @{
+        // Sync
+        @"Sync Enabled": isSyncEnable ? @"true" : @"false",
+        @"Flexible_Sync": isFlexibleSync ? @1 : @0,
```

Comment:
All these places where we're reporting 1/0 instead of "true"/"false" can just be `@(isFlexibleSync)`.

Reply:
This is a requirement on the scope document, all the features should report 1 or 0 if true or not.

Reply:
Yes, and that's what `@(isFlexibleSync)` will do.